yaml-streamly/src/Data/Yaml/Include.hs

```diff
@@ -25,26 +25,30 @@ import Data.Yaml.Internal (ParseException(..), Warning(..), decodeHelper_, decod
 import Text.Libyaml hiding (decodeFile)
 import qualified Text.Libyaml as Y
 
-import           Control.Exception.Safe
-import qualified Streamly.Prelude     as S
-import           Streamly.Prelude (SerialT, MonadAsync)
+import Control.Exception.Safe
 
+import Streamly.Data.Stream (Stream)
+import Streamly.Internal.Data.Stream (CrossStream, unCross, mkCross)
+import qualified Streamly.Data.Stream as S
+
+-- XXX Is there a way we can remove the monadic bind here? And use unfolds for a
+-- concatMap like operations?
```

Comment:
First of all use StreamK instead of Stream because this is a recursive function.
Second, you do not need to use bind, a simple concatMap should be enough here. You can use `append` to append the stream recursively.

Reply:
We won't have to import internal modules here if we use concatMap.